kolibri/plugins/coach/assets/src/composables/useQuizCreation.js

```diff
@@ -323,7 +323,12 @@ export default function useQuizCreation() {
       finalQuiz.question_sources = questionSourcesWithoutResourcePool;
     }
 
-    return ExamResource.saveModel({ id, data: finalQuiz });
+    return ExamResource.saveModel({ id, data: finalQuiz }).then(exam => {
+      if (!get(_quiz).id) {
```

Comment:
Should this maybe be `if (!id || id !== exam.id) {`

`id` here being assigned to the `get(_quiz).id` up above.

Reply:
Presumably if `id` is undefined then it won't equal `exam.id` either, so the falsy check might be redundant.

But yeah, using the id that we already grabbed in the outer scope feels safer, just in case we're accidentally mutating state for some reason.

Reply:
Okay cool - I was getting mixed up a bit around the use of `"new"` for the route until they save the quiz but `get(_quiz).id` will always be `null/undefined` here eh?